pkgs/tools/misc/hdf5/default.nix

```diff
@@ -8,14 +8,18 @@
 , enableShared ? true
 }:
 
-with { inherit (stdenv.lib) optional; };
+# cpp and mpi opitons are mutually exclusive
+# (--enable-unsupported could be used to force the build)
+assert cpp -> mpi == null;
```

Comment:
Yes, both are equivalent, but yours should be preferred because of the symmetric writing. I just amended my commit.